app/models/pet.rb

```diff
@@ -10,4 +12,8 @@ def shelter_name
   def self.adoptable
     where(adoptable: true)
   end
+
+  def self.search(name)
+    where("name ILIKE ?", "%#{name}%")
```

Comment:
Note that this method also exists in the ApplicationRecord file. Since every model inherits from ApplicationRecord, you have access to this method in any model even without declaring it again! You can try this out just by deleting this method and noticing that your application still works, because now your code is going to that parent method declared in ApplicationRecord instead!